chef-deploy.sh

```diff
@@ -43,6 +44,19 @@ wait_for_deployment () {
   fi
 }
 
+find_load_balancer () {
+  # Finds the first load balancer whose tags all match the deployment,
+  # and outputs its identifying name
+  for lb_name in `aws elb describe-load-balancers --query "LoadBalancerDescriptions[].LoadBalancerName" --output text`; do
+    LB_TAGS=$(aws elb describe-tags --load-balancer-name $lb_name --query "TagDescriptions[].Tags[].{Key: Key, Value: Value}[?Key=='Environment' || Key=='Service' || Key == 'Layer']" --output text)
+    if (echo $LB_TAGS | grep -E "Environment\s+$ENVIRONMENT" >/dev/null 2>&1) \
+      && (echo $LB_TAGS | grep -E "Service\s+$SERVICE" >/dev/null 2>&1) \
```

Comment:
should pin with ^ and $ for each line so that it does not match a naughty tag. or a Service that has more data after it.

Reply:
We can only reach this point if the tag values have already matched one or more instances - and those use exact string matches. So it could really only be an issue if we shot ourselves in the foot by making one of our apps or environments a substring of another.